contracts/data/BinaryHeap.sol

```diff
@@ -0,0 +1,320 @@
+    }
+
+    function remove(AddressHeap storage heap, address value)
+        internal
+        returns (bool)
+    {
+        return _remove(heap._inner, bytes32(uint256(uint160(value))));
+    }
+
+    function remove(UintHeap storage heap, uint256 value)
+        internal
+        returns (bool)
+    {
+        return _remove(heap._inner, bytes32(value));
+    }
+
+    function toArray(Bytes32Heap storage heap)
+        internal
+        view
+        returns (bytes32[] memory)
+    {
+        uint256 len = _length(heap._inner);
+        bytes32[] memory arr = new bytes32[](len);
+        for (uint256 index = 0; index < len; index++) {
+            arr[index] = _at(heap._inner, index);
+        }
+        return arr;
+    }
+
+    function toArray(AddressHeap storage heap)
+        internal
+        view
+        returns (address[] memory)
+    {
+        uint256 len = _length(heap._inner);
+        address[] memory arr = new address[](len);
+        for (uint256 index = 0; index < len; index++) {
+            arr[index] = address(uint160(uint256(_at(heap._inner, index))));
+        }
+        return arr;
+    }
+
+    function toArray(UintHeap storage heap)
+        internal
+        view
+        returns (uint256[] memory)
+    {
+        uint256 len = _length(heap._inner);
+        uint256[] memory arr = new uint256[](len);
+        for (uint256 index = 0; index < len; index++) {
+            arr[index] = uint256(_at(heap._inner, index));
+        }
+        return arr;
+    }
+
+    function _at(Heap storage heap, uint256 index)
+        private
+        view
+        returns (bytes32)
+    {
+        require(heap._values.length > index, 'BinaryHeap: index out of bounds');
+        return heap._values[index];
+    }
+
+    function _contains(Heap storage heap, bytes32 value)
+        private
+        view
+        returns (bool)
+    {
+        return heap._indexes[value] != 0;
+    }
+
+    function _indexOf(Heap storage heap, bytes32 value)
+        private
+        view
+        returns (uint256)
+    {
+        unchecked {
+            return heap._indexes[value] - 1;
+        }
+    }
+
+    function _length(Heap storage heap) private view returns (uint256) {
+        return heap._values.length;
+    }
+
+    function _root(Heap storage heap) private view returns (bytes32) {
+        return _at(heap, 0);
+    }
+
+    function _add(Heap storage heap, bytes32 value) private returns (bool) {
+        if (!_contains(heap, value)) {
+            heap._values.push(value);
+            heap._indexes[value] = _length(heap);
+            _heapify(heap);
+
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    function _remove(Heap storage heap, bytes32 value) private returns (bool) {
+        if (_contains(heap, value)) {
+            uint256 index = _indexOf(heap, value);
+
+            // move node with last element in the tree, then remove it
+            _swap(heap, index, _length(heap) - 1);
```

Comment:
I believe this can be `unchecked` because if the `_contains` check passes, the `_length` must be at least 1.